File: app/src/main/java/com/veganbeauty/admin/core/base/UserSession.java
package com.veganbeauty.admin.core.base;

import android.content.Context;
import android.content.SharedPreferences;

public class UserSession {
    private static final String PREFS_NAME = "rootie_admin_prefs";
    private static final String KEY_ROLE = "current_user_role";

    public static String getRole(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        String role = prefs.getString(KEY_ROLE, "admin");
        return role != null ? role : "admin";
    }

    public static void setRole(Context context, String role) {
        SharedPreferences prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        prefs.edit().putString(KEY_ROLE, role).apply();
    }
}

File: app/src/main/java/com/veganbeauty/admin/core/base/UserSession.kt
package com.veganbeauty.admin.core.base

import android.content.Context

object UserSession {
    private const val PREFS_NAME = "rootie_admin_prefs"
    private const val KEY_ROLE = "current_user_role"

    fun getRole(context: Context): String {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        return prefs.getString(KEY_ROLE, "admin") ?: "admin"
    }

    fun setRole(context: Context, role: String) {
        val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        prefs.edit().putString(KEY_ROLE, role).apply()
    }
}
